perun-core/src/main/java/cz/metacentrum/perun/core/impl/AttributesManagerImpl.java

```diff
@@ -3438,6 +3438,17 @@ public void checkAttributeSyntax(PerunSession sess, String key, Attribute attrib
 		attributeModule.checkAttributeSyntax((PerunSessionImpl) sess, key, attribute);
 	}
 
+	@Override
+	public Attribute getAnonymizedValue(PerunSession sess, User user, Attribute attribute) {
+		UserAttributesModuleImplApi attributeModule = getUserAttributeModule(sess, attribute);
+		if (attributeModule == null) {
+			log.debug("Cannot get anonymized attribute value. There is no module for: {}.", attribute.getName());
+			return attribute;
```

Comment:
I think we should be able to anonymize also attributes without attribute modules.

Reply:
I don't think it is necessary. Every configured attribute should have specified its anonymization process in method getAnonymizedValue() - if it is not specified, then the attribute shouldn't be anonymized. What do you think @stavamichal ?

Reply:
I agree with @cuadradek If we need to anonymize specific attributes, we need to create a module for it that specifies how to do that. (the only question here are the core attributes, but we will not anonymize them for now at all - no reason for that so far)
I just changed my mind what is the default. The default shouldn't be null (like it is now) but it should be an old value without change (like in this specific situation). Can you change that? Because if you try to save the same value, we will do nothing (which is perfectly fine here). So if there is no module or the method uses the default setting, you should get the attribute without changed value and you can call setAttribute without any problem.

Reply:
Or you could provide default anonymization method for attributes without module here. You are creating a situation where it is possible to specify (in instance configuration) that certain attributes should be anonymized, but they are not (when called by admins) and the only error message is buried within log file, leaving the admin wondering what has just happened.

Reply:
That's it and the default so far is "no change". It is the same question we are talking about in the comment below this one. The default situation is that there is no way how to the value should be anonymized. And if is there any, you will specify it in the module of the attribute.